### ISSUE TITLE
core dumped error when running more than one character for pxcontrates

### ISSUE THREAD

**Issue by jfwalker** [FePhyFoFum/phyx#24]
EX:  ./pxcontrates -c contrates_file.txt -t ultra_100.tre -a 0

[pxcontrates_example.zip](https://github.com/FePhyFoFum/phyx/files/265248/pxcontrates_example.zip)

**Comment by josephwb** [comment]
Hmm. I can recreate the issue, but don't yet know what is going on; I will track it down today. FWIW, I had made my own dummy examples files for the program the other day, and it runs fine (same tree as in your example):
```
>s10
10.2 4.2
>s9
10.1 4.1
>s8
11.3 5.8
>s7
10.2 4.5
>s6
10.1 4.4
>s5
9.98 4.8
>s4
10.2 4.0
>s3
10.6 6.7
>s2
10.3 4.0
>s1
10.1 5.0
```

**Comment by josephwb** [comment]
Ha. Changing your example data to floats rather than integers makes it work. This seems easy to fix, and reveals a simple bug. Not sure why the first character analysis completes before the crash...

**Comment by josephwb** [comment]
Erg. I swear changing the data to floats worked the other day... Seems like the problem involves `col()` from Armadillo. My guess is that the data from the first character are somehow contaminating the second character (I've run them separately and it works)

**Comment by josephwb** [comment]
Ok, I know what is going on. Will fix this evening.

**Comment by josephwb** [comment]
I give up; it is not the node annotations as I thought. But the first character is definitely contaminating the second character.
Here is the example data I've been using:
tree:
```
((((s1:0.3603553431,s2:0.3603553431):0.8968782862,s3:1.257233629):0.4207546592,(((s4:0.1190332191,s5:0.1190332191):0.02544171746,s6:0.1444749366):0.764764001
9,s7:0.9092389385):0.76874935):0.06349158892,((s8:0.3106889265,s9:0.3106889265):0.1031508249,s10:0.4138397514):1.327640126);
```
data:
```
>s10
10.2 4.2
>s9
10.1 4.1
>s8
11.3 5.8
>s7
10.2 4.5
>s6
10.1 4.4
>s5
9.98 4.8
>s4
10.2 4.0
>s3
10.6 6.7
>s2
10.3 4.0
>s1
10.1 5.0
```

**Comment by josephwb** [comment]
Fixed by @blackrim with 9c1ca2a46878d64c373d2c5f528a5061c8e51198.
